adapters/tlsio_mbedtls.c

```diff
@@ -698,6 +719,77 @@ static void tlsio_mbedtls_DestroyOption(const char* name, const void* value)
     }
 }
 
+int tlsio_mbedtls_setoption(CONCRETE_IO_HANDLE tls_io, const char *optionName, const void *value)
+{
+    int result = 0;
+
+    if (tls_io == NULL || optionName == NULL)
+    {
+        result = __FAILURE__;
+    }
+    else
+    {
+        TLS_IO_INSTANCE *tls_io_instance = (TLS_IO_INSTANCE *)tls_io;
+
+        if (strcmp(OPTION_TRUSTED_CERT, optionName) == 0)
+        {
+            if (tls_io_instance->trusted_certificates != NULL)
+            {
+                // Free the memory if it has been previously allocated
+                free(tls_io_instance->trusted_certificates);
+                tls_io_instance->trusted_certificates = NULL;
+            }
+            if (mallocAndStrcpy_s(&tls_io_instance->trusted_certificates, (const char *)value) != 0)
+            {
+                LogError("unable to mallocAndStrcpy_s");
+                result = __FAILURE__;
+            }
+            else
+            {
+                int parse_result = mbedtls_x509_crt_parse(&tls_io_instance->trusted_certificates_parsed, (const unsigned char *)value, (int)(strlen(value) + 1));
+                if (parse_result != 0)
+                {
+                    LogInfo("Malformed pem certificate");
+                    result = __FAILURE__;
+                }
+                else
+                {
+                    mbedtls_ssl_conf_ca_chain(&tls_io_instance->config, &tls_io_instance->trusted_certificates_parsed, NULL);
+                }
+            }
+        }
+        else if (strcmp(SU_OPTION_X509_CERT, optionName) == 0 || strcmp(OPTION_X509_ECC_CERT, optionName) == 0)
+        {
+            if (mbedtls_x509_crt_parse(&tls_io_instance->owncert, (const unsigned char *)value, (int)(strlen(value) + 1)) != 0)
+            {
+                result = __FAILURE__;
+            }
+            else if (tls_io_instance->pKey.pk_info != NULL)
+            {
+                mbedtls_ssl_conf_own_cert(&tls_io_instance->config, &tls_io_instance->owncert, &tls_io_instance->pKey);
```

Comment:
>mbedtls_ssl_conf_own_cert [](start = 16, length = 25)

Perform error handling if MBEDTLS_ERR_SSL_ALLOC_FAILED (or non zero) is returned

Reply:
fixed